core/drivers/8250_uart.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2015, Linaro Limited
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <platform_config.h>
+
+#include <drivers/8250_uart.h>
+#include <console.h>
+#include <io.h>
+#include <assert.h>
+#include <compiler.h>
+
+/* uart register defines */
+#define UART_RHR	0x0
+#define UART_THR	0x0
+#define UART_IER	0x4
+#define UART_ISR	0x8
+#define UART_FCR	0x8
+#define UART_LCR	0xc
+#define UART_MCR	0x10
+#define UART_LSR	0x14
+#define UART_MSR	0x18
+#define UART_SPR	0x1c
+
+/* uart status register bits */
+#define LSR_TEMT	0x40 /* Transmitter empty */
+#define LSR_THRE	0x20 /* Transmit-hold-register empty */
+#define LSR_EMPTY	(LSR_TEMT | LSR_THRE)
+#define LSR_DR		0x01 /* DATA Ready */
+
+void s8250_uart_init(vaddr_t __unused base,
+		uint32_t __unused uart_clk, uint32_t __unused baud_rate)
+{
+	/* do nothing, debug uart(uart0) share with normal world,
```

Comment:
```
/*
 * Please format multiline comments
 * like this
 */
```

Is it possible to initialize uart here anyway or do we depend on something to be in place before it can be initialized? Without this it's hard to debug the OP-TEE boot.